test/OpenTelemetry.Tests/Internal/JsonStringArrayTagWriterTests.cs

```diff
@@ -0,0 +1,204 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+using System.Text;
+using Xunit;
+
+namespace OpenTelemetry.Internal.Tests;
+
+public class JsonStringArrayTagWriterTests
+{
+    [Theory]
+    [InlineData(new object[] { new char[] { }, "[]" })]
+    [InlineData(new object[] { new char[] { 'a' }, """["a"]""" })]
+    [InlineData(new object[] { new char[] { '1', '2', '3' }, """["1","2","3"]""" })]
+    public void CharArray(char[] data, string expectedValue)
+    {
+        VerifySerialization(data, expectedValue);
+    }
+
+    [Theory]
+    [InlineData(new object[] { new string[] { }, "[]" })]
+    [InlineData(new object[] { new string[] { "one" }, """["one"]""" })]
+    [InlineData(new object[] { new string[] { "" }, """[""]""" })]
+    [InlineData(new object[] { new string[] { "a", "b", "c", "d" }, """["a","b","c","d"]""" })]
+    [InlineData(new object[] { new string[] { "\r\n", "\t", "\"" }, """["\r\n","\t","\u0022"]""" })]
+    [InlineData(new object[] { new string[] { "longlonglonglonglonglonglonglonglong" }, """["longlonglonglonglonglonglonglonglong"]""" })]
+    public void StringArray(string[] data, string expectedValue)
+    {
+        VerifySerialization(data, expectedValue);
+    }
+
+    [Theory]
+    [InlineData(new object[] { new bool[] { }, "[]" })]
+    [InlineData(new object[] { new bool[] { true }, "[true]" })]
+    [InlineData(new object[] { new bool[] { true, false, false, true }, "[true,false,false,true]" })]
+    public void BooleanArray(bool[] data, string expectedValue)
+    {
+        VerifySerialization(data, expectedValue);
+    }
+
+    [Theory]
+    [InlineData(new object[] { new byte[] { }, "[]" })]
+    [InlineData(new object[] { new byte[] { 0 }, "[0]" })]
+    [InlineData(new object[] { new byte[] { byte.MaxValue, byte.MinValue, 4, 13 }, "[255,0,4,13]" })]
```

Comment:
Is this format defined in spec somewhere?

Reply:
Here: https://github.com/open-telemetry/opentelemetry-specification/blob/58bfe48eabe887545198d66c43f44071b822373f/specification/trace/sdk_exporters/zipkin.md?plain=1#L134-L135